internal/pkg/metadb/metadb.go

```diff
@@ -298,7 +302,7 @@ func (m *MetaDB) InsertRecord(ctx context.Context, storeKey string, record *reco
 		}
 		mut := ds.NewInsert(rkey, record)
 		return m.mutateSingleInTransaction(tx, mut)
-	})
+	}, datastore.MaxAttempts(int(m.config.TXMaxRetries)))
```

Comment:
don't you think isn't better to convert the flag from  `uint` to `int` on a higher level so we don't convert it every time ?